### ISSUE TITLE
Ability to control CSS and JS includes within HTML template

### ISSUE THREAD

**Issue by spacedawwwg** [brillout/vite-plugin-ssr#419]
### Description

It would be nice to fine tune the positioning of asset includes within the HTML.

As a front-end developer, I want to be in charge, in very fine detail, where specific tags are included. Currently all CSS is prepended to the `<head>` and scripts appended to `<body>`.

In the case of CSS, the CSS its standard practice to be at least included after a charset meta tag.

In the case of Javascript, there could be blocking scripts that need to be included after the vite compiled injections.

Currently, we currently doing this via a "post-build" script that loops over all the statically generated files - but this is not suitable for any pages using SSR.

### Suggestion

A template tag (e.g. `{{CSS_PLACEHOLDER}}` `{{JS_PLACEHOLDER}}`) that can be utilised within the templates and if they do not exist, fall back to current implementation.

**Comment by brillout** [comment]
How about this?

```js
import { escapeInject, injectAssets, dangerouslySkipEscape } from 'vite-plugin-ssr'

export { render }

function render(pageContext) {
  return escapeInject`<!DOCTYPE html>
    <html>
      <head>
        <meta charset="UTF-8" />
        ${injectAssets(() => pageContext.pageAssets.filter(asset => asset.type==='style'))}
        <title>Vite App</title>
      </head>
      <body>
        <div id="page-view">${dangerouslySkipEscape(pageHtml)}</div>
        ${injectAssets(() => pageContext.pageAssets.filter(asset => asset.type==='script'))}
      </body>
    </html>`
}
```

**Comment by spacedawwwg** [comment]
> How about this?
> 
> ```js
> import { escapeInject, injectAssets, dangerouslySkipEscape } from 'vite-plugin-ssr'
> 
> export { render }
> 
> function render(pageContext) {
>   return escapeInject`<!DOCTYPE html>
>     <html>
>       <head>
>         <meta charset="UTF-8" />
>         ${injectAssets(() => pageContext.pageAssets.filter(asset => asset.type==='style'))}
>         <title>Vite App</title>
>       </head>
>       <body>
>         <div id="page-view">${dangerouslySkipEscape(pageHtml)}</div>
>         ${injectAssets(() => pageContext.pageAssets.filter(asset => asset.type==='script'))}
>       </body>
>     </html>`
> }
> ```

This would be perfect!

**Comment by brillout** [comment]
https://vite-plugin-ssr.com/preload

Released in `0.4.52`.

In case your company is up for it: https://github.com/sponsors/brillout.

Some of what you said is still not possible though.

> In the case of CSS, the CSS its standard practice to be at least included after a charset meta tag.

Why?

> In the case of Javascript, there could be blocking scripts that need to be included after the vite compiled injections.

What's a use case for that?

I originally went for the design of my previous comment, but it didn't work out for various reasons.

**Comment by brillout** [comment]
https://github.com/brillout/vite-plugin-ssr/issues/695
